vscode/src/jdkDownloader.ts

```diff
@@ -189,25 +189,33 @@ export function JDKDownloader(JDKType: string, osType: string, osArchitecture: s
   request.end();
 }
 
-export async function extractJDK(jdkTarballPath: string, extractionTarget: string, jdkVersion: string, osType: string): Promise<void> {
-
-  const extractCommand = `tar -xzf "${jdkTarballPath}" -C "${extractionTarget}"`;
+export async function extractJDK(jdkTarballPath: string, extractionTarget: string, jdkVersion: string, osType: string, jdkType: string): Promise<void> {
+  const downloadedDir = path.join(__dirname, 'jdk_downloads');
+  const extractCommand = `tar -xzf "${jdkTarballPath}" -C "${downloadedDir}"`;
 
+  const oldDirName = `jdk-${jdkVersion}`;
+  const oldDirectoryPath = path.join(downloadedDir, osType === 'macOS' ? oldDirName + '.jdk' : oldDirName);
+  if (fs.existsSync(oldDirectoryPath)) {
+    await fs.promises.rmdir(oldDirectoryPath, { recursive: true });
```

Comment:
I think this has the same problem with deleting pre-existing directories without notification. I wonder: is it possible to create a temporary directy, unpack to it, and then move/rename to the correct location? I think it is OK to remove directories we create (e.g. when something goes wrong), but removing directories that exist without explicit consent seems wrong to me.

Reply:
I have updated the code and following is the flow :
- Oracle JDK or OpenJDK `tar.gz` file will be downloaded
- It would be extracted in a temp folder
- If directory with same name is present in the user selected download location then ask user if they want to delete it or not? 
     - If user agrees for deleting the directory then delete it and move the temp directory to the user selected location with the updated name
     - If user disagrees for deleting the directory then do nothing
- Cleanup the temp folder and the `tar.gz` file